/*
 *  Copyright (C) 2020-2024 Embedded AMS B.V. - All Rights Reserved
 *
 *  This file is part of Embedded Proto.
 *
 *  Embedded Proto is open source software: you can redistribute it and/or 
 *  modify it under the terms of the GNU General Public License as published 
 *  by the Free Software Foundation, version 3 of the license.
 *
 *  Embedded Proto  is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Embedded Proto. If not, see <https://www.gnu.org/licenses/>.
 *
 *  For commercial and closed source application please visit:
 *  <https://EmbeddedProto.com/license/>.
 *
 *  Embedded AMS B.V.
 *  Info:
 *    info at EmbeddedProto dot com
 *
 *  Postal address:
 *    Atoomweg 2
 *    1627 LE, Hoorn
 *    the Netherlands
 */

// This file is used to test including other proto files.

syntax = "proto3";

// Import something simple
import "file_to_include.proto";

// Import a file with nothing in it.
import "empty_file_to_include.proto";

// Import something with variable length
import "repeated_fields.proto";

// Import something from a subfolder.
import "subfolder/file_to_include_from_subfolder.proto";

// Import something from the google folder.
import "google/protobuf/timestamp.proto";


message IncludedMessages {
  some.external.lib.CommonStates  state = 1;
  some.external.lib.CommonMessage msg   = 2;
  repeated_fields rf  = 3;
  sub.package.other_folder_msg sub_msg = 4;
}

// Use a random message from the google file.
message TimeMessage {
  google.protobuf.Timestamp time = 1;
}